amplipi/rt.py

```diff
@@ -63,6 +64,15 @@ def __init__(self, mock=False):
       self.bus = None
       print('Mocking preamp connection')
     else:
+      # Reset preamp board before establishing a communication channel
+      GPIO.setmode(GPIO.BCM)
+      GPIO.setup(4, GPIO.OUT)
+      GPIO.output(4, 1)
```

Comment:
A couple things here.
- Do we really need to set NRST high before pulling low? There is an internal 40k pull-up in the STM32 so it should already be high.
  - NRST becomes an output
- These times all seem super long.
  - 100ms for resetting seems excessive, 1ms should be more than enough. The datasheet isn't very clear on how long is required but seems to imply >300ns would work.
  - Is the 1 second mentioned the time it takes for all 6 units to restart? Even assuming they reset in sequence that's 167ms per unit, if they are taking that long to be ready to talk to something is wrong.
- Is there a way to check for the serial connection to be ready? Try sending a test message or something? Then we don't have to figure out some max wait time, we just keep retrying at a fast interval until it's good-to-go. Do we ever use the UART's RX (to the Pi)? If not we could even use that as GPIO to signal ready.